### PR TITLE
make preview intpromote the default

### PR DESCRIPTION
**Pull request by adamdruppe**

### CODE REVIEW

**Comment by adamdruppe** [comment]
walter said on the livestream he would do it now

so let's do it now!!!!

**Comment by ibuclaw** [comment]
I think you have to do a bit more than this.

What about deprecating the command line option?

Removing the parameter and conditional uses of it!

File: src/dmd/globals.h [hunk 1]
```diff
@@ -137,7 +137,7 @@ struct Param
     bool betterC;       // be a "better C" compiler; no dependency on D runtime
     bool addMain;       // add a default main() function
     bool allInst;       // generate code for all template instantiations
-    bool fix16997;      // fix integral promotions for unary + - ~ operators
+    bool fix16997 = true;      // fix integral promotions for unary + - ~ operators
```
Comment (dkorpel) [inline src/dmd/globals.h:140]: This is only the .h file, it needs to change in globals.d

**Comment by adamdruppe** [comment]
yeah i don't actually know this process just like i want action done

**Comment by ljmf00** [comment]
I believe we also need to amend the specification: https://dlang.org/spec/type.html#integer-promotions

**Comment by Geod24** [comment]
For context: https://issues.dlang.org/show_bug.cgi?id=20678
Integral promotion is really a dumb rule, and only works because C is permissive. I'm really not looking forward to this being merged.

**Comment by adamdruppe** [comment]
I agree, I hate the whole int promote nonsense and I'd prefer to be rid of it.

But if it is what is going to happen anyway, the compiler can at least stop nagging me all the time. This obnoxious nagging and incessant casts is awful. (And this switch doesn't actually help that much, but at least it stops complaining about ~ushort. D's rules here - and yes i know it is copying C's rules plus explicit casts but still - are asinine and I find it hard to believe anyone who actually uses them finds positive value in them.)

**Comment by 12345swordy** [comment]
@adamdruppe address @dkorpel review please.

**Comment by Imperatorn** [comment]
Anyway.

Let's say this gets merged (after fixup).

How could one invert it later?

**Comment by ibuclaw** [comment]
> Anyway.
> 
> Let's say this gets merged (after fixup).
> 
> How could one invert it later?

This is what `-revert=` is for.

**Comment by Imperatorn** [comment]
@ibuclaw Yes, but for how long

**Comment by adamdruppe** [comment]
https://github.com/dlang/dmd/pull/13348

**Comment by 12345swordy** [comment]
@ibuclaw I talk to Adam in the discord he is not interested in advancing this PR anymore, as frustrated with the leadership that @WalterBright has shown. Please see here Iain https://github.com/dlang/dmd/pull/13348

**Comment by ibuclaw** [comment]
> @ibuclaw Yes, but for how long

Long enough.

**Pull request closed without merging**